### PR TITLE
fix(nav): Fix bug where previous was expanding on select

### PR DESCRIPTION
**Pull request by ibolton336**
**What**:
#1016 

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Nav/NavToggle.js [hunk 1]
```diff
@@ -18,29 +18,30 @@ class NavToggle extends React.Component {
   static propTypes = propTypes;
   static defaultProps = defaultProps;
   state = {
-    value: this.props.defaultValue
+    toggleValue: this.props.defaultValue
   };
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.defaultValue !== this.state.value) {
-      this.setState({ value: nextProps.defaultValue });
+  componentDidUpdate(prevProps) {
```
Comment (priley86) [inline packages/patternfly-4/react-core/src/components/Nav/NavToggle.js]: 👍

**Comment by priley86** [comment]
LGTM - 🐛 🔨 

after

File: packages/patternfly-4/react-core/src/components/Nav/NavToggle.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const propTypes = {
 };
 
 const defaultProps = {
-  defaultValue: true,
+  defaultvalue: false,
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Nav/NavToggle.js]: Not camel case

**Comment by rhamilto** [comment]
This fix introduces some new bugs:

1.  Upon app load, active sections are not expanded.

yhnpvoa2v6

2.  Previously expanded items that have been collapsed items are (once again) not re-expanded.

ixkroplivy

cc: @spadgett, @janwright73, @priley86

**Comment by spadgett** [comment]
It seems like the underlying problem is that we're tracking the same value using both props and state, which is generally considered an anti-pattern. Would it be simpler just to let the outer component control expanded state and add a callback for when the nav section is activated? Then the expanded prop is the single source of truth.

It puts more work on the outer component, but I don't see a good way around that as long as the expanded state can change based on things that happen outside the nav (such as the user clicking on a link on a details page).

**Comment by priley86** [comment]
this should mostly be resolved now. One step closer here...
https://github.com/openshift/console/pull/1238#issuecomment-469763337

**Pull request merged**